### DISCUSSION TITLE
Tweepy only pulls the last day's worth of data.

### DISCUSSION THREAD

**Discussion by reisner** [tweepy/tweepy#1378] [Questions] [answered]
Hi there,

I've been using a cursor with `api.search` and it has been running fine for months, but since around april it seems to only pull a single day's worth of data, instead of the usual (~up to one week). Did something change in tweepy, or in the twitter API?

I'm using:
```
for tweet in tweepy.Cursor(api.search, q = query, rpp = 100, since_id = 1, tweet_mode = "extended").items():
```

When `query` is something that doesnt return too many results, like `query = "#yegtransit"`, I get 7 days worth of results. When I use something that gives more results, like `query = "#yeg"`, it only returns one day of results.

**Comment by reisner** [comment]
So it looks like the API changed and now uses `count` instead of `rpp`. After changing to this, it works:

```
for tweet in tweepy.Cursor(api.search, q = query, count = 100, since_id = 1, tweet_mode = "extended").items():
```

I get the same behaviour (i.e. returning only one day) when i dont specify either `count` or `rpp`. Why does not setting the results-per-page mean that I'm changing how far back I get results?

**Comment by Harmon758** [comment, marked as answer]
`API.search` uses [the standard search API](https://developer.twitter.com/en/docs/twitter-api/v1/tweets/search/api-reference/get-search-tweets), for which `rpp` hasn't been a supported parameter for years.

As for receiving less results when `count` isn't specified, you're likely hitting the rate limit sooner, since you'd only be getting the default 15 results per request rather than 100.
